File: R/Endpoint Examples/JSON Payload/blank-pdf.R
#! 
# What this sample does:
# - Calls /blank-pdf with a JSON payload to create an empty three-page PDF.
#
# Setup (.Renviron):
# - Copy .Renviron.example to .Renviron (R folder root)
# - Set PDFREST_API_KEY=your_api_key_here
# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
#     PDFREST_URL=https://eu-api.pdfrest.com
#   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
#
# Usage:
#   Rscript "Endpoint Examples/JSON Payload/blank-pdf.R"
#
# Output:
# - Prints the API JSON response to stdout. Non-2xx responses quit with a concise message.

suppressWarnings(suppressMessages({
  if (!requireNamespace("httr", quietly = TRUE)) stop("Please install 'httr' package")
  if (!requireNamespace("jsonlite", quietly = TRUE)) stop("Please install 'jsonlite' package")
}))

stderrf <- function(...) cat(sprintf(...), file = stderr())

api_key <- Sys.getenv("PDFREST_API_KEY", unset = "")
if (identical(api_key, "")) {
  stderrf("Missing PDFREST_API_KEY in environment (.Renviron or shell)\n")
  quit(status = 1)
}

api_base <- sub("/+$", "", Sys.getenv("PDFREST_URL", unset = "https://api.pdfrest.com"))

payload <- jsonlite::toJSON(
  list(page_size = "letter", page_count = 3, page_orientation = "portrait"),
  auto_unbox = TRUE
)

resp <- httr::POST(
  paste0(api_base, "/blank-pdf"),
  httr::add_headers(
    "api-key" = api_key,
    "Content-Type" = "application/json"
  ),
  body = payload
)

body_text <- httr::content(resp, as = "text", encoding = "UTF-8")
cat(body_text)

if (httr::http_error(resp)) {
  stderrf("\nblank-pdf request failed with status %s\n", httr::status_code(resp))
  quit(status = 1)
}

File: Perl/Endpoint Examples/JSON Payload/blank-pdf.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;
use FindBin qw($Bin);
use JSON::PP qw(encode_json);
use LWP::UserAgent;
use HTTP::Request;
use Dotenv;

#!
# What this sample does:
# - Requests a three-page blank PDF using a JSON payload.
#
# Setup (.env):
# - Copy .env.example to .env (Perl folder root)
# - Set PDFREST_API_KEY=your_api_key_here
# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
#     PDFREST_URL=https://eu-api.pdfrest.com
#   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
#
# Usage:
#   perl "Endpoint Examples/JSON Payload/blank-pdf.pl"
#
# Output:
# - Prints the API JSON response to stdout. Non-2xx responses exit non-zero.

binmode STDOUT, ':raw';
binmode STDERR, ':encoding(UTF-8)';

my $env_path = "$Bin/../../.env";
-e $env_path and Dotenv->load($env_path);

my $api_key = $ENV{PDFREST_API_KEY} // '';
if (!$api_key || $api_key =~ /^\s*$/) {
    print STDERR "Missing PDFREST_API_KEY in .env or environment\n";
    exit 1;
}

my $api_base = $ENV{PDFREST_URL} // $ENV{PDFREST_API} // 'https://api.pdfrest.com';
$api_base =~ s{/+$}{};

my $ua = LWP::UserAgent->new( timeout => 60 );

my $payload = encode_json({
    page_size  => 'letter',
    page_count => 3,
    page_orientation => 'portrait',
});

my $request = HTTP::Request->new('POST', "$api_base/blank-pdf");
$request->header('api-key' => $api_key);
$request->header('Content-Type' => 'application/json');
$request->content($payload);

my $response = $ua->request($request);
print STDOUT $response->decoded_content // '';

unless ($response->is_success) {
    print STDERR "\nblank-pdf request failed with status " . $response->code . "\n";
    exit 1;
}

exit 0;

__END__
